In [140]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [141]:
df=pd.read_csv("D:\\medical-11\\data\\diet_recommendations_dataset.csv")

In [142]:
df.head()

,Patient_ID,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
0,P0001,56,Male,58.4,160,22.8,Obesity,Moderate,Moderate,3079,173.3,133,116.3,NaN,Peanuts,Mexican,3.1,96.6,3.1,Balanced
1,P0002,69,Male,101.2,169,35.4,Diabetes,Mild,Moderate,3032,199.2,120,137.1,NaN,Peanuts,Chinese,4.5,63.2,0.6,Low_Carb
2,P0003,46,Female,63.5,173,21.2,Hypertension,Mild,Sedentary,1737,181.0,121,109.6,NaN,Peanuts,Chinese,3.8,57.5,4.6,Low_Sodium
3,P0004,32,Male,58.1,164,21.6,NaN,Mild,Moderate,2657,168.2,144,159.4,NaN,NaN,Mexican,4.3,54.5,0.4,Balanced
4,P0005,60,Male,79.5,197,20.5,Diabetes,Moderate,Sedentary,3496,200.4,172,182.3,Low_Sugar,NaN,Italian,9.8,78.2,4.7,Low_Carb


droping the patient id as it is not important

In [143]:
df.drop(columns=['Patient_ID'],inplace=True)

In [144]:
df.head()

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
0,56,Male,58.4,160,22.8,Obesity,Moderate,Moderate,3079,173.3,133,116.3,NaN,Peanuts,Mexican,3.1,96.6,3.1,Balanced
1,69,Male,101.2,169,35.4,Diabetes,Mild,Moderate,3032,199.2,120,137.1,NaN,Peanuts,Chinese,4.5,63.2,0.6,Low_Carb
2,46,Female,63.5,173,21.2,Hypertension,Mild,Sedentary,1737,181.0,121,109.6,NaN,Peanuts,Chinese,3.8,57.5,4.6,Low_Sodium
3,32,Male,58.1,164,21.6,NaN,Mild,Moderate,2657,168.2,144,159.4,NaN,NaN,Mexican,4.3,54.5,0.4,Balanced
4,60,Male,79.5,197,20.5,Diabetes,Moderate,Sedentary,3496,200.4,172,182.3,Low_Sugar,NaN,Italian,9.8,78.2,4.7,Low_Carb


In [145]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 19 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Age                               1000 non-null   int64  
 1   Gender                            1000 non-null   object 
 2   Weight_kg                         1000 non-null   float64
 3   Height_cm                         1000 non-null   int64  
 4   BMI                               1000 non-null   float64
 5   Disease_Type                      796 non-null    object 
 6   Severity                          1000 non-null   object 
 7   Physical_Activity_Level           1000 non-null   object 
 8   Daily_Caloric_Intake              1000 non-null   int64  
 9   Cholesterol_mg/dL                 1000 non-null   float64
 10  Blood_Pressure_mmHg               1000 non-null   int64  
 11  Glucose_mg/dL                     1000 non-null   float64
 12  Dietary

In [146]:
df.shape

(1000, 19)

In [147]:
df.duplicated().sum()

np.int64(0)

In [148]:
df.isnull().sum()

Age                                   0
Gender                                0
Weight_kg                             0
Height_cm                             0
BMI                                   0
Disease_Type                        204
Severity                              0
Physical_Activity_Level               0
Daily_Caloric_Intake                  0
Cholesterol_mg/dL                     0
Blood_Pressure_mmHg                   0
Glucose_mg/dL                         0
Dietary_Restrictions                334
Allergies                           323
Preferred_Cuisine                     0
Weekly_Exercise_Hours                 0
Adherence_to_Diet_Plan                0
Dietary_Nutrient_Imbalance_Score      0
Diet_Recommendation                   0
dtype: int64

diesease type have 204 missing vale 
dietary restriction \
\allergies have too

In [149]:
df['Age'].describe()

count    1000.000000
mean       49.857000
std        18.114267
min        18.000000
25%        35.000000
50%        50.000000
75%        66.000000
max        79.000000
Name: Age, dtype: float64

In [150]:
df['Gender'].value_counts()

Gender
Male      523
Female    477
Name: count, dtype: int64

In [151]:
df['Gender']=df['Gender'].map({'Male':1, 'Female':0}).astype(int)

In [152]:
df['Gender'].value_counts()

Gender
1    523
0    477
Name: count, dtype: int64

In [153]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 19 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Age                               1000 non-null   int64  
 1   Gender                            1000 non-null   int64  
 2   Weight_kg                         1000 non-null   float64
 3   Height_cm                         1000 non-null   int64  
 4   BMI                               1000 non-null   float64
 5   Disease_Type                      796 non-null    object 
 6   Severity                          1000 non-null   object 
 7   Physical_Activity_Level           1000 non-null   object 
 8   Daily_Caloric_Intake              1000 non-null   int64  
 9   Cholesterol_mg/dL                 1000 non-null   float64
 10  Blood_Pressure_mmHg               1000 non-null   int64  
 11  Glucose_mg/dL                     1000 non-null   float64
 12  Dietary

weight_kg

In [154]:
df['Weight_kg'].describe()

count    1000.000000
mean       84.602400
std        20.088121
min        50.000000
25%        66.600000
50%        85.200000
75%       102.000000
max       119.700000
Name: Weight_kg, dtype: float64

In [155]:
train_df=df.copy()

In [156]:
train_df['Weight_kg'].isnull().sum()

np.int64(0)

In [157]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
604,70,0,117.3,180,36.2,NaN,Moderate,Sedentary,2968,199.3,172,109.1,NaN,NaN,Chinese,5.9,66.6,1.9,Balanced
658,40,1,51.6,183,15.4,Hypertension,Severe,Moderate,2910,165.5,136,101.3,Low_Sugar,NaN,Indian,8.3,50.2,2.2,Low_Sodium


Height_cm

In [158]:
train_df['Height_cm'].describe()

count    1000.00000
mean      174.81700
std        14.33377
min       150.00000
25%       162.00000
50%       175.00000
75%       187.00000
max       199.00000
Name: Height_cm, dtype: float64

BMI

In [159]:
train_df['BMI'].describe()

count    1000.000000
mean       28.191600
std         8.040136
min        13.000000
25%        22.075000
50%        27.450000
75%        33.425000
max        52.400000
Name: BMI, dtype: float64

Disease_Type

In [160]:
train_df['Disease_Type'].value_counts()

Disease_Type
Hypertension    316
Diabetes        258
Obesity         222
Name: count, dtype: int64

In [161]:
train_df['Disease_Type'].isnull().sum()

np.int64(204)

In [162]:
train_df[train_df['Disease_Type'].isna()]

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
3,32,1,58.1,164,21.6,NaN,Mild,Moderate,2657,168.2,144,159.4,NaN,NaN,Mexican,4.3,54.5,0.4,Balanced
6,78,1,102.2,170,35.4,NaN,Mild,Active,2879,175.8,166,95.1,NaN,Gluten,Chinese,9.2,56.9,1.6,Balanced
14,41,1,83.7,154,35.3,NaN,Moderate,Moderate,2896,230.8,122,138.5,NaN,Gluten,Chinese,9.3,98.3,1.1,Balanced
20,39,0,106.3,198,27.1,NaN,Severe,Active,3297,214.6,177,93.8,Low_Sodium,Peanuts,Indian,6.2,72.0,0.4,Balanced
21,70,1,87.7,168,31.1,NaN,Moderate,Sedentary,1583,195.8,138,126.1,Low_Sodium,Peanuts,Mexican,1.4,85.4,0.1,Balanced
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
956,67,1,53.9,153,23.0,NaN,Mild,Active,1938,154.1,118,182.2,Low_Sugar,NaN,Mexican,4.1,89.2,2.2,Balanced
973,30,0,59.2,190,16.4,NaN,Mild,Sedentary,3048,242.5,140,143.1,NaN,Gluten,Indian,8.7,52.6,0.5,Balanced
974,63,0,77.9,184,23.0,NaN,Moderate,Moderate,2643,201.7,142,157.2,NaN,Gluten,Italian,1.7,92.8,0.7,Balanced
976,78,0,83.0,197,21.4,NaN,Moderate,Sedentary,1852,202.8,142,107.9,Low_Sodium,Gluten,Mexican,1.9,54.5,3.6,Balanced


replacing the nan value of the diesease into no diabetes as for now

In [163]:
train_df['Disease_Type']=train_df['Disease_Type'].fillna('Normal')

In [164]:
train_df['Disease_Type'].value_counts()

Disease_Type
Hypertension    316
Diabetes        258
Obesity         222
Normal          204
Name: count, dtype: int64

In [165]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
636,31,1,66.7,158,26.7,Normal,Moderate,Moderate,3347,213.7,124,131.3,Low_Sugar,Gluten,Italian,6.1,85.1,4.3,Balanced
893,79,1,58.7,160,22.9,Diabetes,Severe,Moderate,2792,177.8,166,146.2,NaN,Gluten,Italian,3.2,81.4,3.0,Low_Carb


# Severity
define as the  stage he/she is curently on 

In [166]:
train_df['Severity'].value_counts()

Severity
Severe      344
Moderate    342
Mild        314
Name: count, dtype: int64

In [167]:
train_df['Severity'].isnull().sum()

np.int64(0)

In [168]:
train_df['Severity']=train_df['Severity'].map({'Mild':0, 'Moderate':1, 'Severe':2}).astype(int)

In [169]:
train_df['Severity'].value_counts()

Severity
2    344
1    342
0    314
Name: count, dtype: int64

In [170]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
122,42,1,96.3,195,25.3,Obesity,1,Moderate,1959,154.9,169,129.3,NaN,Gluten,Mexican,8.9,50.5,3.4,Balanced
80,64,0,106.1,170,36.7,Normal,2,Active,2015,232.6,138,160.1,NaN,Gluten,Mexican,5.2,58.9,0.3,Balanced


Physical_Activity_Level

In [171]:
train_df['Physical_Activity_Level'].value_counts()

Physical_Activity_Level
Moderate     335
Active       335
Sedentary    330
Name: count, dtype: int64

In [172]:
train_df['Physical_Activity_Level'].isnull().sum()

np.int64(0)

In [173]:
train_df['Physical_Activity_Level']=train_df['Physical_Activity_Level'].map({'Sedentary':1.2, 'Moderate':1.55, 'Active':1.725}).astype('float')

In [174]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
55,54,0,53.2,151,23.3,Obesity,2,1.725,2392,223.6,176,178.5,Low_Sodium,Gluten,Mexican,0.3,52.3,2.9,Balanced
337,24,0,105.5,199,26.6,Hypertension,0,1.200,2744,213.0,170,189.8,NaN,NaN,Indian,2.1,89.4,2.3,Low_Sodium


# Daily_Caloric_Intake

In [175]:
train_df['Daily_Caloric_Intake'].describe() 

count    1000.000000
mean     2475.064000
std       565.017032
min      1500.000000
25%      1984.750000
50%      2470.500000
75%      2937.250000
max      3498.000000
Name: Daily_Caloric_Intake, dtype: float64

In [176]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
308,34,1,80.4,179,25.1,Hypertension,2,1.2,3343,166.7,148,77.1,Low_Sodium,Peanuts,Mexican,9.8,82.4,0.2,Low_Sodium
53,74,0,69.4,171,23.7,Hypertension,2,1.2,3141,221.0,174,142.5,NaN,NaN,Mexican,2.5,79.0,4.8,Low_Sodium


In [177]:
train_df.sample(2)

,Age,Gender,Weight_kg,Height_cm,BMI,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Dietary_Restrictions,Allergies,Preferred_Cuisine,Weekly_Exercise_Hours,Adherence_to_Diet_Plan,Dietary_Nutrient_Imbalance_Score,Diet_Recommendation
559,32,0,64.4,152,27.9,Hypertension,0,1.55,1864,183.3,158,81.0,Low_Sugar,NaN,Chinese,9.2,96.8,0.1,Low_Sodium
506,43,0,86.6,159,34.3,Diabetes,0,1.20,3062,212.7,175,131.5,NaN,NaN,Mexican,2.1,99.3,2.3,Low_Carb


In [178]:
df.drop(columns=['Adherence_to_Diet_Plan','BMI','Allergies','Preferred_Cuisine','Dietary_Restrictions','Dietary_Nutrient_Imbalance_Score'],inplace=True)

In [179]:
df.head()

,Age,Gender,Weight_kg,Height_cm,Disease_Type,Severity,Physical_Activity_Level,Daily_Caloric_Intake,Cholesterol_mg/dL,Blood_Pressure_mmHg,Glucose_mg/dL,Weekly_Exercise_Hours,Diet_Recommendation
0,56,1,58.4,160,Obesity,Moderate,Moderate,3079,173.3,133,116.3,3.1,Balanced
1,69,1,101.2,169,Diabetes,Mild,Moderate,3032,199.2,120,137.1,4.5,Low_Carb
2,46,0,63.5,173,Hypertension,Mild,Sedentary,1737,181.0,121,109.6,3.8,Low_Sodium
3,32,1,58.1,164,NaN,Mild,Moderate,2657,168.2,144,159.4,4.3,Balanced
4,60,1,79.5,197,Diabetes,Moderate,Sedentary,3496,200.4,172,182.3,9.8,Low_Carb


In [180]:
df.isnull().sum()

Age                          0
Gender                       0
Weight_kg                    0
Height_cm                    0
Disease_Type               204
Severity                     0
Physical_Activity_Level      0
Daily_Caloric_Intake         0
Cholesterol_mg/dL            0
Blood_Pressure_mmHg          0
Glucose_mg/dL                0
Weekly_Exercise_Hours        0
Diet_Recommendation          0
dtype: int64

In [181]:
df['Disease_Type']=df['Disease_Type'].fillna('Normal')

In [182]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Age                      1000 non-null   int64  
 1   Gender                   1000 non-null   int64  
 2   Weight_kg                1000 non-null   float64
 3   Height_cm                1000 non-null   int64  
 4   Disease_Type             1000 non-null   object 
 5   Severity                 1000 non-null   object 
 6   Physical_Activity_Level  1000 non-null   object 
 7   Daily_Caloric_Intake     1000 non-null   int64  
 8   Cholesterol_mg/dL        1000 non-null   float64
 9   Blood_Pressure_mmHg      1000 non-null   int64  
 10  Glucose_mg/dL            1000 non-null   float64
 11  Weekly_Exercise_Hours    1000 non-null   float64
 12  Diet_Recommendation      1000 non-null   object 
dtypes: float64(4), int64(5), object(4)
memory usage: 101.7+ KB


In [183]:
df.to_csv("D:\\medical-11\\data\\processed_diet_recommendations_dataset.csv",index=False)